fhttp/http_test.go

```diff
@@ -941,6 +941,28 @@ func TestChangeMaxPayloadSize(t *testing.T) {
 	}
 }
 
+func TestAddHTTPS(t *testing.T) {
+	var tests = []struct {
+		input    string
+		expected string
+	}{
+		{"foo.com", "https://foo.com"},
+		{"http://foo.com", "https://foo.com"},
+		{"https://foo.com", "https://foo.com"},
+		{"hTTps://foo.com", "https://foo.com"},
```

Comment:
add /BaR there to confirm when it's already https it's also preserving the case
(and/or fOO.com)